README.rst

```diff
@@ -63,13 +65,27 @@ Settings list
 -------------
 All settings are contained in a ``PUSH_NOTIFICATIONS_SETTINGS`` dict.
 
-In order to use GCM, you are required to include ``GCM_API_KEY``.
-For APNS, you are required to include ``APNS_CERTIFICATE``.
+In order to use GCM, you are required to include one of ``GCM_API_KEY``, ``GCM_API_KEYS``, or ``GCM_API_KEYS_MODEL``.
+For APNS, you are required to include ``APNS_CERTIFICATE``, ``APNS_CERTIFICATES``, or ``APNS_CERTIFICATES_MODEL``.
 
 - ``APNS_CERTIFICATE``: Absolute path to your APNS certificate file. Certificates with passphrases are not supported.
+- ``APNS_CERTIFICATES``: A dictionary reflecting separate application IDs to separate APNS certificate files.
+- ``APNS_CERTIFICATES_MODEL``: A dictionary containing description of a database model to reflect
```

Comment:
**I swear** that I'm not trying to derail this PR attempt in favour of the one (#339) I submitted, I'm truly good with whatever is considered best for this project. Obviously, I'm coming from my use case where I want to support apps on any platform `push_notifications` supports (apns, gcm, fcm, wns) and benefit from any future platforms added (web push, etc?) as well as any data modelling changes needed, so bear that in mind. With that being said... 

I'm not sure that having a configurable 'Application' model makes the most sense for `push_notifications`. 

With this configurable model, the onus is on the developer using `push_notifications` to update their custom 'Application' models to add any additional fields for future platform additions (or any where the field length is increased, for example).

Also, keep in mind that the platform messaging logic is a part of the `push_notifications` app, but by abstracting away the certificate storage logic it becomes somewhat of a support nightmare - everyone has a different implementation based on loose conventions.

By having the model 'in-house' (a part of the `push_notifications` project itself), these scenarios are no longer problems. Messaging logic is tightly coupled to known entities. Migrations can take care of both new platform additions as well as data model changes to existing platforms.

If the developer using `push_notifications` needs to associate additional data with an 'Application' model instance, he/she can use relations (ie. `OneToOneField`, `ForeignKey`).

If configurable models are considered necessary, then I would highly suggest following the pattern Django uses with `AUTH_USER_MODEL`: the standard Django `User` model is built upon a set of 'mixins' and this standard `User` model is the default if the developer does not specify `AUTH_USER_MODEL` in their settings.

Although I don't consider it necessary for this project (I think a single non-configurable model would suffice), consider this pattern for `push_notifications`:

```
# default model setting
PUSH_NOTIFICATIONS_APPLICATION_MODEL = 'push_notifications.Application'

# developers could choose to derive their own model from this to add 
# additional non-push_notification fields
class AbstractBaseApplication(models.Model):
    # all the apns, fcm/gcm, wns etc fields (and add future platform 
    #fields here)
    ...
    
    class Meta:
        abstract = True

# this would be the default model used, unless 
# PUSH_NOTIFICATIONS_APPLICATION_MODEL was specified in Django 
# project settings
class Application(AbstractBaseApplication):
    pass
```

I think these points are fairly important to consider.